From 5de1730d857f8496f6a72244bbc4c3736078bca5 Mon Sep 17 00:00:00 2001
From: Arnav Gupta <championswimmer@gmail.com>
Date: Sat, 6 Feb 2016 22:15:14 +0530
Subject: [PATCH 25/44] minui: accept RGBA and treat it as RGBX

There are devices with RGBA_8888 pixel format.

There are also recoveries like TWRP that have
alpha support and hence for a RGBA device, I expect
to be able to set TARGET_RECOVERY_PIXEL_FORMAT = RGBA_8888

It should be supported on BoardConfig to set the
correct PixelFormat for a device.
Although recovery does not have alpha, so it can
consider RGBA devices as RGBX, but it could do so
in future.

Change-Id: Id1f3b2fbfc8ad4cf0422614580e420196614f792
Signed-off-by: Arnav Gupta <championswimmer@gmail.com>
---
 minui/Android.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/minui/Android.mk b/minui/Android.mk
index f33fde53..72f46662 100644
--- a/minui/Android.mk
+++ b/minui/Android.mk
@@ -51,6 +51,9 @@ endif
 ifeq ($(subst ",,$(TARGET_RECOVERY_PIXEL_FORMAT)),RGBX_8888)
   LOCAL_CFLAGS += -DRECOVERY_RGBX
 endif
+ifeq ($(subst ",,$(TARGET_RECOVERY_PIXEL_FORMAT)),RGBA_8888)
+  LOCAL_CFLAGS += -DRECOVERY_RGBX
+endif
 ifeq ($(subst ",,$(TARGET_RECOVERY_PIXEL_FORMAT)),BGRA_8888)
   LOCAL_CFLAGS += -DRECOVERY_BGRA
 endif
-- 
2.17.1

